File: LeetCode/504-Base 7/solution.cpp
// Author: David
// Email: youchen.du@gmail.com
// Created: 2017-02-16 08:55
// Last modified: 2017-02-16 08:55
// Filename: solution.cpp
// Description:
class Solution {
public:
    string convertToBase7(int num) {
        string sign;
        string digits="";
        int mod;
        if(num > 0)
            sign = "";
        else if(num < 0)
            sign = "-";
        else
            return "0";
        num = abs(num);
        while(num != 0)
        {
            mod = num % 7;
            num = num / 7;
            digits = to_string(mod)+digits;
        }
        return sign+digits;
    }
};

File: LeetCode/504-Base 7/solution.py
#!/usr/local/bin/python3
# coding: UTF-8
# Author: David
# Email: youchen.du@gmail.com
# Created: 2017-02-16 08:51
# Last modified: 2017-02-16 08:51
# Filename: solution.py
# Description:
class Solution(object):
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        if num > 0:
            sign = ''
        elif num < 0:
            sign = '-'
        else:
            return '0'
        digits = ''
        num = abs(num)
        while num != 0:
            num, m = int(num / 7), num % 7
            digits += str(m)
        return sign+digits[::-1]
